packages/one-bx-shell-app/src/app/components/plugin-launcher/plugin-launcher.component.ts

```diff
@@ -74,37 +70,31 @@ export class PluginLauncherComponent implements OnChanges {
       }
       this.descriptor = item.descriptor;
     }
-    this.doCreateComponent(this.descriptor);
+    this.doCreateComponent(this.descriptor, this.module);
   }
 
 
-  async doCreateComponent(descriptor: Partial<ExtendedPluginDescriptor>): Promise<void> {
-    const component = await loadRemoteModule(descriptor)
-      .then(m => m[descriptor.componentName]);
+  async doCreateComponent(descriptor: Partial<ExtendedPluginDescriptor>, moduleName: string): Promise<void> {
+    const _module = descriptor.modules.find(module => module.name === moduleName);
+    const _component = await loadRemoteModule<AngularIvyComponentDescriptor>(descriptor, _module as AngularIvyComponentDescriptor)
+      .then(m => m[_module.name]);
 
-    if (descriptor.framework === 'legacy') {
-      const element = document.createElement(component);
+    if (_module.type === 'custom-element') {
```

Comment:
Should we have some enumeration for different types? so we can use enum type instead of 'xxx'

Reply:
Module types are easier with strings literals